### PR TITLE
Refactor rankmap

### PR DESCRIPTION
**Pull request by yfguo**
## Pull Request Description

The rankmap is not necessarily device specific. This PR moves it to
MPIR_ for better decoupling. This will eventually decouple the rankmap
creation with `MPID_Comm_create_hook`.

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [ ] Commits are self-contained and do not do two things at once
* [ ] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Passes whitespace checkers
* [ ] Passes warning tests
* [ ] Passes all tests
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -605,7 +605,10 @@ Makefile.am-stamp
 # /test/mpi/
 /test/mpi/libtool
 /test/mpi/coll/testlist.cvar
+<<<<<<< HEAD
 /test/mpi/attr/testlist.dtp
 /test/mpi/pt2pt/testlist.dtp
 /test/mpi/coll/testlist.dtp
 /test/mpi/rma/testlist.dtp
+=======
+>>>>>>> update gitignore for new generate files.
```
Comment (hzhou) [inline .gitignore]: Oops, bad conflict resolution.

**Comment by yfguo** [comment]
test:jenkins/ch4/ofi

**Comment by yfguo** [comment]
test:jenkins/ch4/ofi

**Comment by yfguo** [comment]
@yfguo Need some minor fixes and cleanup.

**Comment by yfguo** [comment]
test:jenkins/ch4/ofi

**Comment by yfguo** [comment]
test:jenkins/ch3/tcp

**Comment by pavanbalaji** [comment]
@yfguo Is this PR needed or can we close it?

**Comment by yfguo** [comment]
We don't need this any more.

**Pull request closed without merging**